src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let stateClone = { ...state };
+  const list = [];
+
+  function removeProperties(obj, keysToRemove) {
+    for (let index = 0; index < keysToRemove.length; index++) {
+      delete stateClone[keysToRemove[index]];
+      delete obj[keysToRemove[index]];
+    }
+
+    return obj;
```

Comment:
The 'removeProperties' function should not mutate 'stateClone' directly. It should only work with the 'obj' parameter that is passed to it. This will ensure that the original 'stateClone' is not affected by the removal of properties.